### DISCUSSION TITLE
Using the Auth module: What is the best way to change a password?

### DISCUSSION THREAD

**Discussion by agedeo** [leafsphp/leaf#247] [Q&A] [answered]
I am using the Auth module. What is the best way to change a users password on the users request?

I have a form with the old password and the new password.

How do I verify that the old password is the correct password and how do I update the password in the database?

**Comment by ibnsultan** [comment, marked as answer]
You can use the [Leaf Password](https://github.com/leafsphp/password) module to do that, which is aleady loaded as a Helper class

assuming you're using a leaf mvc, a rough implementtion would be something like this

```php
use App\Models\User;
use Leaf\Helpers\Password;

$oldPassword = 'theOldPass';
$newPassword = 'theNewPass';

$user = User::find(:id);
if(Password::verify($oldPassword, $user->password){
    $user->password = Password::hash($newPassword)
    $user->save();
}
```
